### PR TITLE
Docs: Update comma-style docs

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

**Comment by kaicataldo** [comment]
@pedrottimark I was working on a fix for this rule and noticed the docs had some confusing language. Thought I could freshen/clean it up a bit. Your thoughts? And if this steps on your toes with anything you're doing, I'm happy to close it -- appreciate all the work you've been doing on this!

File: docs/rules/comma-style.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Comma style (comma-style)
 
-Comma Style rule enforces comma styles for a list of things separated by commas. There are two comma styles primarily in JavaScript. The standard one in which commas are placed at the end of the line. And Comma-First, in which, commas are placed at the start of the next line after the list item.
+The Comma Style rule enforces styles for comma-separated lists. There are two comma styles primarily used in JavaScript. The standard one in which commas are placed at the end of the line, and Comma First, in which commas are placed at the start of the next line after the list item.
```
Comment (pedrottimark) [inline docs/rules/comma-style.md]: Would it help or hurt to reorganize the information in the third sentence as two items in a bulleted list?

Comment (kaicataldo) [inline docs/rules/comma-style.md]: I think it could definitely help if done the right way! I can give it a shot if you're onboard.

Comment (pedrottimark) [inline docs/rules/comma-style.md]: Go for it 😀

**Comment by pedrottimark** [comment]
Forgot to mention a general principle which this rule doc provides a positive example:
* Introduction section explains **what** is the software quality goal, **why** or **to whom** it matters.
* Rule Details and Options explains **how** to use the rule to achieve the goal.

In some rule docs, introduction leans forward into how and Rule Details leans backward into why. Here are two items in my mental check list:
* Make sure an answer is clear in context if the reader asks “why is the doc telling me this?”
* Reorganize fuzzy repetitions which could cause readers to ask “is the doc repeating the same thing or explaining something else?”

**Comment by kaicataldo** [comment]
@pedrottimark Updated :)

**Comment by kaicataldo** [comment]
@pedrottimark Do we have a guide for writing docs? Could be really helpful to have info like the comment above somewhere in our [Working with rules](http://eslint.org/docs/developer-guide/working-with-rules) docs!

**Comment by kaicataldo** [comment]
@pedrottimark Let me know what you think - I'm pretty happy with where it is now :)

**Comment by pedrottimark** [comment]
looks good to me 👍

**Pull request merged**